packages/app/background-jobs-common/src/background-job-processor/processors/AbstractBackgroundJobProcessor.ts

```diff
@@ -41,7 +41,14 @@ export abstract class AbstractBackgroundJobProcessor<
   JobReturn = void,
   ExecutionContext = undefined,
   JobType extends SafeJob<JobPayload, JobReturn> = Job<JobPayload, JobReturn>,
-  QueueType extends Queue<JobPayload, JobReturn> = Queue<JobPayload, JobReturn>,
+  QueueType extends Queue<JobPayload, JobReturn, string, JobPayload, JobReturn, string> = Queue<
+    JobPayload,
+    JobReturn,
+    string,
+    JobPayload,
+    JobReturn,
+    string
+  >,
```

Comment:
As `JobPayload` is not defined at this stage, TS can't infer `NameType` at this stage 
```
export declare class Queue<..., DefaultNameType extends string = string, ..., NameType extends string = ExtractNameType<DataTypeOrJob, DefaultNameType>> {

type ExtractNameType<DataTypeOrJob, Default extends string> = DataTypeOrJob extends Job<any, any, infer N> ? N : Default;
```

So trying to define it explicitly to help TS